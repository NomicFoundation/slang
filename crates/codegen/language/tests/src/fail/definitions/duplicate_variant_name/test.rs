#![allow(unused_crate_dependencies)]

codegen_language_macros::compile!(Language(
    name = Foo,
    documentation_dir = "foo/bar",
    root_item = Bar,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [Topic(
            title = "Topic One",
            items = [
                Enum(
                    name = Bar,
                    variants = [
                        EnumVariant(reference = Baz1),
                        EnumVariant(reference = Baz2),
                        EnumVariant(reference = Baz1)
                    ]
                ),
                Token(
                    name = Baz1,
                    definitions = [TokenDefinition(scanner = Atom("baz1"))]
                ),
                Token(
                    name = Baz2,
                    definitions = [TokenDefinition(scanner = Atom("baz2"))]
                )
            ]
        )]
    )],
    queries = ()
));

fn main() {}
