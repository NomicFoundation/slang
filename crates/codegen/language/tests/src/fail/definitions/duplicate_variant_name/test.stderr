error: A variant referencing 'Baz1' already exists.
  --> src/fail/definitions/duplicate_variant_name/test.rs:19:49
   |
19 |                         EnumVariant(reference = Baz1)
   |                                                 ^^^^
